tests/test_utils.go

```diff
@@ -335,7 +336,31 @@ func removeURLSecrets(u *url.URL) string {
 
 // MatchInteraction checks if the request matches a store request in the given cassette.
 func MatchInteraction(r *http.Request, i cassette.Request) bool {
-	return r.Method == i.Method && removeURLSecrets(r.URL) == i.URL
+	// Default matching on method and URL without secrets
+	if !(r.Method == i.Method && removeURLSecrets(r.URL) == i.URL) {
+		return false
+	}
+
+	if r.Body == nil {
```

Comment:
What if `r.Body == nil`, but `i.Body != nil`? Shouldn't this condition only return true if bot `r.Body` and `i.Body` are nil at the same time?